Alexandria/boardgame-backend
# Feature/bycrypt

## Reference

[37](https://crema.atlassian.net/browse/BA-37)

## Overview

I created a sign up path in my api that encrypts the users password before it is put into the database.
I modified the login path to check to see if entered password matches the encrypted one  in the data base.   

### Acceptance Criteria

- [x] Passwords are no longer stored in plain text in the database

### Demo

#### Screenshot of the users in the data base.
<img width="1237" alt="Screen Shot 2019-07-01 at 11 57 00 AM" src="https://user-images.githubusercontent.com/7613670/60453675-624dbd00-9bf7-11e9-8243-bc5d05a3ee65.png">


## Reviewers' Checklist (go ahead, check 'em)


## Testing Instructions

Before testing this build you will need to have postgres installed and have a database already created.
Please look at the readme to see how to set up your environment variables.

1. Checkout branch feature/bcrypt
2. If you have tables built in your database run `npx sequelize db:migrate:undo:all`, if not skip this step
3. Run `npx sequelize db:migrate`
4. Run `npx sequelize db:seed:all`
5. Log in at http://localhost:YOUR_PORT/auth/login
6. Email: admin@gmail.com,  password:password
7. You can create your own user at http://localhost:YOUR_PORT/auth/signup
